sympy/functions/elementary/hyperbolic.py

```diff
@@ -624,6 +625,38 @@ def as_real_imag(self, deep=True, **hints):
         denom = sinh(re)**2 + cos(im)**2
         return (sinh(re)*cosh(re)/denom, sin(im)*cos(im)/denom)
 
+    def _eval_expand_trig(self, **hints):
+        arg = self.args[0]
+        x = None
+        if arg.is_Add:
+            from sympy import symmetric_poly
+            n = len(arg.args)
+            TX = []
+            if arg.is_Add:
+                for x in arg.args:
+                    tx = tanh(x, evaluate=False)._eval_expand_trig()
+                    TX.append(tx)
+
+                Yg = numbered_symbols('Y')
+                Y = [ next(Yg) for i in range(n) ]
+
+                p = [0, 0]
+                for i in range(n + 1):
+                    p[1 - i % 2] += symmetric_poly(i, Y)
+                return (p[0]/p[1]).subs(list(zip(Y, TX)))
```

Comment:
```suggestion
                return (p[0]/p[1]).xreplace(dict(zip(Y, TX)))
```

Reply:
We could probably avoid substitutions by using `TX` instead of `Y` in `symmetric_polynomial`. There may be repetitions in `TX` but that could be handled by adding `repetition=True` when calling subsets: https://github.com/sympy/sympy/blob/8f3790a92fe47d33477ebead73f218fadb839d9d/sympy/polys/specialpolys.py#L107

Reply:
This would make symmetric polynomials less minimal, e.g. `symmetric_polynomial(2, x, y)` would become `x**2 + x*y + y**2` instead of `x*y`. `symmetric_polynomial(2, x, x, y)` would be `3*x**2 + 2*x*y + y**2` but could be `x*y`, too.

Reply:
Yes, my mistake. `symmetric_poly` seems to have no problems with repeated arguments:
```
In [19]: symmetric_poly(2, x, x, y)                                             
Out[19]: 
 2        
x  + 2⋅x⋅y
```
That could be used with `TX` instead of `Y`.